invoiceregistry_test.go

```diff
@@ -170,3 +172,56 @@ func TestInvoiceExpiry(t *testing.T) {
 	}
 	cancel3()
 }
+
+func TestAutoSettle(t *testing.T) {
+	c := newRegistryTestContext(t)
+
+	// Subscribe to updates for invoice 1.
+	updateChan, cancelUpdates := c.subscribe(1)
+
+	// Add invoice.
+	c.addInvoice(1, time.Hour, true)
+
+	// Expect an open notification.
+	update := <-updateChan
+	require.Equal(t, persistence.InvoiceStateOpen, update.State)
+
+	preimage := c.preimage(1)
+	resolved := make(chan struct{})
+	c.registry.NotifyExitHopHtlc(&registryHtlc{
+		rHash:         preimage.Hash(),
+		amtPaid:       lnwire.MilliSatoshi(c.testAmt),
+		expiry:        100,
+		currentHeight: 0,
+		resolve: func(r HtlcResolution) {
+			c.logger.Infow("resolution", "res", r)
+			close(resolved)
+		},
+		payload: &testPayload{
+			amt:     lnwire.MilliSatoshi(c.testAmt),
+			payAddr: c.payAddr(1),
+		},
+	})
+
+	update = <-updateChan
+	require.Equal(t, persistence.InvoiceStateAccepted, update.State)
+
+	update = <-updateChan
+	require.Equal(t, persistence.InvoiceStateSettleRequested, update.State)
+
+	update = <-updateChan
+	require.Equal(t, persistence.InvoiceStateSettled, update.State)
+
+	<-resolved
+
+	cancelUpdates()
```

Comment:
Invoked via `defer` for consistency (or `t.Cleanup`) ?

Reply:
In `TestInvoiceExpiry`, I cancel them without a defer because there are multiple subscriptions that need to be cancelled before the test finishes. Therefore I also didn't use defer in this new test.

Reply:
the fact that is not defered suggests that it will be cancelled somewhere in the middle of the test.
If you are going to call cancel at the end, why not prefer defer?

Reply:
Seems better to me for consistency with the other test.

Reply:
ok